contracts/upgradeable_contracts/BaseFeeManager.sol

```diff
@@ -0,0 +1,65 @@
+pragma solidity 0.4.24;
+
+import "../upgradeability/EternalStorage.sol";
+import "../libraries/SafeMath.sol";
+import "../IRewardableValidators.sol";
+
+
+contract BaseFeeManager is EternalStorage {
+    using SafeMath for uint256;
+
+    event FeeUpdated(uint256 fee);
+
+    function calculateFee(uint256 _value, bool _recover) external payable returns(uint256) {
```

Comment:
I had to set this method as `payable` otherwise it will revert when trying to call it using `delegatecall` from fallback method on `HomeBridgeErcToNative` https://github.com/poanetwork/poa-bridge-contracts/blob/c7d62b43e1097a2211c808a5e1430c436756813c/contracts/upgradeable_contracts/erc20_to_native/HomeBridgeErcToNative.sol#L28

Reply:
Thanks for suggestion! `callcode` worked https://github.com/poanetwork/poa-bridge-contracts/pull/132/commits/98dfc8becfaf7aa4fbd2ae2b9440c0032cc3293c